db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java

```diff
@@ -92,27 +92,43 @@ public void check( LanguageConnectionContext lcc,
 		DataDictionary dd =	lcc.getDataDictionary();
 		TransactionController tc = lcc.getTransactionExecute();
         String currentUserId = lcc.getCurrentUserId(activation);
+		List<String> currentGroupuserlist = lcc.getCurrentGroupUser(activation);
 
 		SchemaDescriptor sd;
 		switch ( privType )
 		{
+			case Authorizer.ACCESS_PRIV:
+				/* for access priv, all the permissions set schemaUUID instead of schemaName */
+				sd = dd.getSchemaDescriptor(schemaUUID, tc);
+				if (sd == null)
+					return;
+
+				// if current user is owner, it can access the schema
+				if (currentUserId.equals(sd.getAuthorizationId()) ||
+						currentGroupuserlist != null && currentGroupuserlist.contains(sd.getAuthorizationId()))
+					return;
+
+				if (!hasPermissionOnSchema(lcc, dd, activation, forGrant))
+					throw StandardException.newException(
+							SQLState.LANG_SCHEMA_DOES_NOT_EXIST,
+							sd.getSchemaName());
+				break;
 			case Authorizer.MODIFY_SCHEMA_PRIV:
 				sd = dd.getSchemaDescriptor(schemaName, tc, false);
 				if (sd == null)
 					return;
 
 				// if current user is owner, it can modify the schema
-                if (currentUserId.equals(sd.getAuthorizationId()))
+                if (currentUserId.equals(sd.getAuthorizationId()) ||
```

Comment:
nit: duplication and not an easy one to understand so I'd request a separate method that would return a boolean value `true` when access is granted and `false` otherwise, e.g. `isAuthorized`?

Reply:
This is consistent with code in other modules, so I will keep it this way.